test/ExtractGoal.lean

```diff
@@ -52,3 +58,82 @@ example {z : Int} : ∃ n : Nat, ↑n = z := by
   set_option pp.all true in
   extract_goal
   sorry
+
+/--
+info: theorem foo : True := sorry
+---
+warning: declaration uses 'sorry'
+-/
+#guard_msgs in
+example (n : ℕ) : True := by
+  extract_goal using foo
+  sorry
+
+/--
+info: theorem foo (n : ℕ) : True := sorry
+---
+warning: declaration uses 'sorry'
+-/
+#guard_msgs in
+example (n : ℕ) : True := by
+  extract_goal n using foo
+  sorry
+
+/-- error: unknown identifier 'k' -/
+#guard_msgs in
+example (n : ℕ) : True := by
+  extract_goal k
+
+/--
+info: theorem extracted_1 (n : ℕ) : True := sorry
+---
+warning: declaration uses 'sorry'
+-/
+#guard_msgs in
+example (n : ℕ) : True := by
+  extract_goal *
+  sorry
+
+-- Clears `i` neither `n` nor the goal depends on it.
```

Comment:
```suggestion
-- Clears `i` since neither `n` nor the goal depend on it.
```

Reply:
Or some equivalent: the current phrasing confused me.

Reply:
Thanks, added "since". (The rule I know is that "neither" is always singular in formal writing, so I left "depends".)